deployment/ccip/changeset/cs_configure_token_admin_registry.go

```diff
@@ -0,0 +1,299 @@
+			}
+		}
+	}
+
+	proposeAdminOutput, err := deployerGroup.Enact(fmt.Sprintf("propose admin for %s on token admin registries", c.TokenSymbol))
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("propose admin for %s on token admin registries: %w", c.TokenSymbol, err)
+	}
+
+	// Accept admin pass
+	for chainSelector := range c.RegistryUpdates {
+		cc := chainConfigs[chainSelector]
+
+		if cc.TokenConfigOnRegistry.Administrator != cc.Sender {
+			_, err := cc.State.TokenAdminRegistry.AcceptAdminRole(cc.Opts, cc.TokenAddress)
+			if err != nil {
+				return deployment.ChangesetOutput{}, fmt.Errorf("failed to create acceptAdminRole transaction for %s on %s registry: %w", c.TokenSymbol, cc.Chain, err)
+			}
+		}
+	}
+
+	acceptAdminOutput, err := deployerGroup.Enact(fmt.Sprintf("accept admin rights for %s on token admin registries", c.TokenSymbol))
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to accept admin rights for %s on token admin registries: %w", c.TokenSymbol, err)
+	}
+
+	// Configuration pass (set pool, transfer admin role to 3rd party)
+	for chainSelector, registryUpdate := range c.RegistryUpdates {
+		cc := chainConfigs[chainSelector]
+
+		// Only set the pool if we need to
+		if cc.TokenConfigOnRegistry.TokenPool != cc.TokenPool.Address() {
+			_, err := cc.State.TokenAdminRegistry.SetPool(cc.Opts, cc.TokenAddress, cc.TokenPool.Address())
+			if err != nil {
+				return deployment.ChangesetOutput{}, fmt.Errorf("failed to create setPool transaction for %s on %s registry: %w", c.TokenSymbol, cc.Chain, err)
+			}
+		}
+
+		// Only set the administrator to an external address if we need to
+		if registryUpdate.ExternalAdministrator != cc.Sender {
+			_, err := cc.State.TokenAdminRegistry.TransferAdminRole(cc.Opts, cc.TokenAddress, registryUpdate.ExternalAdministrator)
+			if err != nil {
+				return deployment.ChangesetOutput{}, fmt.Errorf("failed to create transferAdminRole transaction for %s on %s registry: %w", c.TokenSymbol, cc.Chain, err)
+			}
+		}
+	}
+
+	configurationOutput, err := deployerGroup.Enact(fmt.Sprintf("configure %s on token admin registries", c.TokenSymbol))
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to configure %s on token admin registries: %w", c.TokenSymbol, err)
+	}
+
+	if c.MCMS != nil {
+		// Pre-allocate the proposal slice with the correct capacity
+		totalProposals := len(proposeAdminOutput.Proposals) +
+			len(acceptAdminOutput.Proposals) +
+			len(configurationOutput.Proposals)
+		proposals := make([]timelock.MCMSWithTimelockProposal, 0, totalProposals)
+		proposals = append(proposals, proposeAdminOutput.Proposals...)
+		proposals = append(proposals, acceptAdminOutput.Proposals...)
+		proposals = append(proposals, configurationOutput.Proposals...)
+
+		return deployment.ChangesetOutput{
+			Proposals: proposals,
+		}, nil
+	}
```

Comment:
One thing I have been thinking is that since the deployerGroup was initially made to abstract MCMS vs deployer key. Would it make sense to also move this logic into the deployerGroup?

Perhaps we could have a function called .NewProposal(reason) instead of using Enact in multi proposal scenario. Not sure if this is needed for this PR but interested in your opinion if that would be a good next step

Reply:
I think revising the API so that we return a single `Enact` at the end of the changeset makes the most sense.